### ISSUE TITLE
ContentDialog default buttons do not respect ControlCornerRadius setting

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#2134]
**Describe the bug**
If you set the `ControlCornerRadius` on the application level, only the default ContentDialog button respects this setting. All other in-built dialog buttons ignore this setting and still present the default 2px corner radius.

I can create a repro if requested, but this can also be put together real quick. Just show a ContentDialog with at least one button which is not the default button and put 
```
<CornerRadius x:Key="OverlayCornerRadius">0,0,0,0</CornerRadius>
```
in the application's resources below the WinUI's resource dictionary.

**Expected behavior**
All ContentDialog buttons have sharp corners.

**Screenshots**
![image](https://user-images.githubusercontent.com/1398851/77099654-992d6280-6a14-11ea-88ea-a9bd98c7e993.png)

**Version Info**
NuGet package version: 
Microsoft.UI.Xaml 2.3.200213001 and newest 2.4 pre-release

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (19577)              | Yes   |
| November 2019 Update (18363)  | Yes   |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**
I would like to give this a shot and fix it (or try for a workaround in WinUI if the error cause is i the framework). Unfortunately, I'm not sure how I would test my code with just the MUXControlsTestApp. The CornerRadius page's ContentDialog test is broken (the ControlCornerRadius values are not applied). See the following image for example:
![radius issue](https://user-images.githubusercontent.com/1398851/77105471-009be000-6a1e-11ea-97e8-c9c8c8f46f9a.png)

As they are buttons inside a ContentDialog which is (like) a Popup, I believe you can only set the ControlCornerRadius resource on the application level to affect these buttons. However, adding 
```
<CornerRadius x:Key="ControlCornerRadius">0</CornerRadius>
```
at the bottom of App.xaml for the MUXControlsTestApp ([here](https://github.com/microsoft/microsoft-ui-xaml/blob/master/test/MUXControlsTestApp/App.xaml#L38)) does not change any value as it would in a plain UWP project created with VS.

As such, the CornerRadius test page should be fixed to also make testing ControlCornerRadius work for ContentDialogs. I would be open to lok into this issue as well.

Also note that some issues related to resources defined on the application level seem to have been fixed in recent Insider builds. Not sure if this is the case here as well and compiling with an Insider SDK applies the ControlCornerResource correctly to all buttons in a ContentDialog.

**Comment by chingucoding** [comment]
Regarding testing this: Can one define a ContentDialog in XAML or only code behind? Either way, you should be able specify the resources of the ContentDialog and with that be able to set ControlCornerRadius value.

**Comment by Felix-Dev** [comment]
@Chingucoding Yes, you can define a ContentDialog in XAML and then show it in code-behind:
```xml
<ContentDialog x:Name="ExampleDialog"
                Title="Some Title"
                Content="Some Content"
                PrimaryButtonText="PrimaryButton"
                SecondaryButtonText="SecondaryButton"
                CloseButtonText="CoseButton"
                DefaultButton="Close">
    <ContentDialog.Resources>
        <CornerRadius x:Key="ControlCornerRadius">0</CornerRadius>
        <CornerRadius x:Key="OverlayCornerRadius">20</CornerRadius>
    </ContentDialog.Resources>
</ContentDialog>
```
and in code-behind:
```
this.ExampleDialog.ShowAsync();
```

note that setting the ControlCornerRadius value in the ContentDialog resources doesn't work though (OverlayCornerRadius works):
![image](https://user-images.githubusercontent.com/1398851/77107818-8a997800-6a21-11ea-8457-463a294200bb.png)

**Comment by chingucoding** [comment]
That is weird. Why does setting it locally in the ContentDialog itsself not work 🤔
I guess the only way to test this then, is to have a a line in the App.xaml file we comment/uncomment for testing.

**Comment by Felix-Dev** [comment]
I added the fact that the specific issue about (ControlCornerRadius only appiled to the defaut button) only happens when the dialog is fully created in *code-behind*. If you create the dialog in XAML, not even the default button respects this setting. This could be related to issue https://github.com/microsoft/microsoft-ui-xaml/issues/1486.

**Comment by chingucoding** [comment]
The "code behind" situation is probably caused by the issue you linked. However the fact that only one of the buttons respects the theme resource, is probably caused by something else. If the only button respecting the resource is the button set as default, you might want to look at 
https://github.com/microsoft/microsoft-ui-xaml/blob/440966762fcb622e12a8e6f596514a355d04739a/dev/ContentDialog/ContentDialog_themeresources.xaml#L242-L259

and the way the buttons are defined in ContentDialog.

**Comment by Felix-Dev** [comment]
@chingucoding I took a look previously and it looks fine to me. Both the DefaultButtonStyle and the AccentButtonStyle (used by ContentDialog's default button) reference the `ControlCornerRadius` resource correctly. One difference between both button styles in the context of ContentDialog is the following I believe: The AccentButtonStyle is set later as the DefaultButtonStyle as it is only set in a VisualState. By default, all buttons (even the default one) initially get the DefaultButtonStyle.

Now I wanted to test this by swapping both the DefaultButtonStyle and AccentButtonStyle with each other (so the default button would use the DefaultButtonStyle set in a VisualState, while the other buttons would use the AccentButtonStyle) and see if then the default button using the DefaultButtonStyle would also respect the ControlCornerRadius value whereas the buttons usig the AccentButtonStyle would not. However, I'm not aware of any way right now to test this reliably using just WinUI (remember the testing issues I mentioned above). As such, I could not yet test if this difference in timing for both the DefaultButtonStyle and AccentButtonStyle would be the reason why the resource is respected by the button using the AccentButtonStyle and not by the others.

(Hopefully you could follow me here 😅)

**Comment by chingucoding** [comment]
That's a good guess. The main difference is between the default and not default button is that the default button has a different style specified than initially specified in template. My guess would be that the CloseButtonStyle, PrimaryButtonStyle and SecondaryButtonStyle are the problem.
However I don't know where those are defined 😅

**Comment by Felix-Dev** [comment]
Default values for the styles you mention are listed here (they are using the DefaultButtonStyle I referenced in the post above): https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/ContentDialog/ContentDialog_themeresources.xaml#L103

**Comment by chingucoding** [comment]
Thanks for the link. Since the DefaultButtonStyle should work correctly, the last thing you could try is setting the CornerRadius explicitly for the buttons. If that doesn't work, it probably has something to do with timing and might be unfixable (as #1486).

**Comment by StephenLPeters** [comment]
also seems related to #1345.

**Comment by hawkerm** [comment]
Still seeing square buttons in the Content Dialog:

![image](https://user-images.githubusercontent.com/8959496/210468788-77a5cc5f-df05-43e4-b66b-53023d2c4853.png)

Tried overriding it in my Content Dialog's XAML:

```xml
<ContentDialog.Resources>
        <CornerRadius x:Key="ControlCornerRadius">4</CornerRadius>
```

But that didn't change anything.

In fact if I look at the properties for the Button itself, it seems like the Style itself isn't properly set at all:

![image](https://user-images.githubusercontent.com/8959496/210468898-f5f32419-5bbb-4765-a576-265de682c934.png)

So, it's not even trying to set the Corner Radius or use the `ControlCornerRadius` value to begin with.

If I instead, manually re-define the style to be the `DefaultButtonStyle` (which is supposed to be its default from the template)

```xml
<ContentDialog x:Class="..." PrimaryButtonStyle="{ThemeResource DefaultButtonStyle}">
```

https://github.com/microsoft/microsoft-ui-xaml/blob/84bc7ac6672a879ed9e7d879bb142b716d97507b/dev/CommonStyles/ContentDialog_themeresources.xaml#L71-L77

Then I get the expected behavior:

![image](https://user-images.githubusercontent.com/8959496/210469018-84058054-a6e0-4288-be82-8021f6c2ec5f.png)

Could it be the default `ContentDialog` template (or something with how defining a custom ContentDialog) is causing the ContentDialog style itself not to be picked up?

![image](https://user-images.githubusercontent.com/8959496/210469325-737fbab9-0c58-40f8-a5a1-6f2487a49343.png)

**Comment by MartinZikmund** [comment]
@hawkerm Have you tried just setting:

```
Style="{ThemeResource DefaultContentDialogStyle}"
```

This should apply all styling, so you can avoid styling the buttons explicitly.

**Comment by niels9001** [comment]
> [@hawkerm](https://github.com/hawkerm) Have you tried just setting:
> 
> ```
> Style="{ThemeResource DefaultContentDialogStyle}"
> ```
> 
> This should apply all styling, so you can avoid styling the buttons explicitly.

This. Setting the `DefaultContentDialogStyle` is required to ensure the modern styles are applied to `ContentDialog`
